scripts/install-deps.sh

```diff
@@ -0,0 +1,210 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+ARG=""
+DIRECTORY="/tmp"
+
+#######################################
+# Feature flags
+#######################################
+
+# If true → do not download anything (unless custom URL is used)
+DISABLE_EXTERNAL_DOWNLOADS="${DISABLE_EXTERNAL_DOWNLOADS:-false}"
+
+# Optional custom URL prefix for all binaries
+CUSTOM_KUBECTL_URL="${CUSTOM_KUBECTL_URL:-}"
+CUSTOM_JQ_URL="${CUSTOM_JQ_URL:-}"
+CUSTOM_OC_URL="${CUSTOM_OC_URL:-}"
+
+#######################################
+# OS / ARCH Detection
+#######################################
+
+if [[ $OSTYPE == 'darwin'* ]]; then
+	OS="darwin"
+	# Detect Apple Silicon vs Intel
+	mac_arch="$(sysctl -a 2>/dev/null | grep machdep.cpu.brand_string || true)"
+	if [[ "$mac_arch" == 'machdep.cpu.brand_string: Intel'* ]]; then
+		ARCH="amd64"
+	else
+		ARCH="arm64"
+	fi
+else
+	OS="linux"
+	ARCH="amd64"
+fi
+
+#######################################
+# Download helpers
+#######################################
+
+function download {
+	local binary=$1
+	local version=$2
+	local url=$3
+	local file=$4
+	local tmp_dir=$5
+	local custom_url=$6
+
+	if [[ "$DISABLE_EXTERNAL_DOWNLOADS" == "true" && "$custom_url" != "true" ]]; then
+		echo "Downloads disabled (DISABLE_EXTERNAL_DOWNLOADS=true). Skipping $binary."
+		return
+	fi
+
+	if [ "$custom_url" = "true" ]; then
+		echo "Downloading ${binary}..."
+		curl --retry 3 -fLsS "${url}" --output "${tmp_dir}/${file}"
+	else
+		echo "Downloading ${binary} ${version}..."
+		curl --retry 3 -fLsS "${url}/${file}" --output "${tmp_dir}/${file}"
+	fi
+}
+
+#######################################
+# Permission check
+#######################################
+
+function permission_check {
+	if ! [ -w "${DIRECTORY}" ]; then
+		echo "No write permission to ${DIRECTORY}. Using sudo..."
+		ARG=sudo
+	fi
+}
+
+#######################################
+# Install to DIRECTORY
+#######################################
+
+function copy_replace_binary {
+	local binary=$1
+	local tmp_dir=$2
+
+	echo "Placing ${binary} into ${DIRECTORY}..."
+	permission_check
+
+	${ARG} rm -f "${DIRECTORY}/${binary}"
+	${ARG} cp -r "${tmp_dir}/${binary}" "${DIRECTORY}"
+	${ARG} chmod +x "${DIRECTORY}/${binary}"
+}
+
+#######################################
+# Cleanup
+#######################################
+
+function clean {
+	local tmp_dir=$1
+	echo "Deleting tmp dir: ${tmp_dir}"
+	rm -rf "${tmp_dir}"
+	echo "COMPLETE"
+	echo
+}
+
+#######################################
+# Install: kubectl
+#######################################
+
+# renovate: datasource=github-releases depName=kubernetes/kubernetes
```

Comment:
you will need to add a custom renovate rule in renovate.json for this to work (copy the one in common-dev-asstes)